src/main/java/org/fcrepo/storage/ocfl/PersistencePaths.java

```diff
@@ -0,0 +1,250 @@
+     * @param resourceId the id of the acl resource to get the paths for
+     * @return paths
+     */
+    public static PersistencePaths aclResource(final boolean describesRdfResource,
+                                               final String rootResourceId, final String resourceId) {
+        final var info = analyze(rootResourceId, resourceId);
+        if (!info.isAcl) {
+            throw new IllegalArgumentException("This function should only be called for ACL resources.");
+        }
+        final var headerPath = resolveHeaderPath(info);
+        final var contentPath = resolveContentPath(describesRdfResource, info);
+        return new PersistencePaths(contentPath, headerPath);
+    }
+
+    private static String resolveContentPath(final boolean isContainer, final IdInfo info) {
+        if (info.isRoot) {
+            if (isContainer) {
+                if (info.isAcl) {
+                    return CONTAINER_PREFIX + ACL_SUFFIX + RDF_EXTENSION;
+                } else {
+                    return CONTAINER_PREFIX + RDF_EXTENSION;
+                }
+            }
+        }
+
+        final var pathBuilder = new StringBuilder(info.relativeId);
+
+        if (info.isDescription) {
+            pathBuilder.append(DESCRIPTION_SUFFIX).append(RDF_EXTENSION);
+        } else if (isContainer) {
+            if (info.isAcl) {
+                pathBuilder.append("/").append(CONTAINER_PREFIX).append(ACL_SUFFIX).append(RDF_EXTENSION);
+            } else {
+                pathBuilder.append("/").append(CONTAINER_PREFIX).append(RDF_EXTENSION);
+            }
+        } else if (info.isAcl) {
+            pathBuilder.append(ACL_SUFFIX).append(RDF_EXTENSION);
+        }
+
+        return pathBuilder.toString();
+    }
+
+    private static String resolveHeaderPath(final IdInfo info) {
+        String path;
+
+        if (info.isRoot) {
+            path = ROOT_PREFIX;
+        } else {
+            path = info.relativeId;
+        }
+
+        if (info.isAcl) {
+            path += ACL_SUFFIX;
+        } else if (info.isDescription) {
+            path += DESCRIPTION_SUFFIX;
+        }
+
+        return headerPath(path);
+    }
+
+    private static String headerPath(final String path) {
+        return HEADER_DIR + path + JSON_EXTENSION;
+    }
+
+    private static IdInfo analyze(final String rootResourceId, final String resourceId) {
+        if (rootResourceId.equals(resourceId)) {
+            return IdInfo.root(resourceId);
+        }
+
+        if (!resourceId.startsWith(rootResourceId)) {
+            throw new IllegalArgumentException(String.format("The resources %s and %s are unrelated",
+                    resourceId, rootResourceId));
+        }
+
+        final var relative = resourceId.substring(rootResourceId.length() + 1);
+
+        if (relative.equals(FCR_ACL)) {
+            return IdInfo.rootAcl(resourceId.substring(0, resourceId.lastIndexOf('/')));
```

Comment:
Its pretty much the same thing, but seems like you could use `StringUtils.substringBeforeLast(resourceId, '/')` for the sake of readability if the commons libraries are already available. Just a suggestion since this same operation is happening in so many places, I don't feel strongly about it.